src/transport/emitter.ts

```diff
@@ -131,4 +147,48 @@ export class Emitter {
     }
     return this.binaryEmitter(event, options);
   }
+
+  /**
+   * Return or create the Emitter singleton
+   *
+   * @return {Emitter} return Emitter singleton
+   */
+  static getInstance(): Emitter {
+    if (!Emitter.instance) {
+      Emitter.instance = new Emitter();
+    }
+    return Emitter.instance;
+  }
+
+  /**
+   * Add a listener for eventing
+   *
+   * @param {string} event type to listen to
+   * @param {Function} listener to call on event
+   * @return {void}
+   */
+  static on(event: "cloudevent" | "newListener" | "removeListener", listener: (...args: any[]) => void): void {
+    this.getInstance().on(event, listener);
+  }
+
+  /**
+   * Emit an event inside this application
+   *
+   * @param {CloudEvent} event to emit
+   * @param {boolean} ensureDelivery fail the promise if one listener fail
+   * @return {void}
+   */
+  static async emitEvent(event: CloudEvent, ensureDelivery = false): Promise<void> {
+    if (!ensureDelivery) {
+      // Ensure delivery is disable so we do wait for Promise
+      Emitter.getInstance().emit("cloudevent", event);
+    } else {
+      // Execute all listeners and wrap them in a Promise
+      await Promise.all(
+        Emitter.getInstance()
+          .listeners("cloudevent")
+          .map((l) => Promise.resolve(l(event))),
```

Comment:
This looks like it will never reject. Shouldn't this be something more like this?
```suggestion
          .map((l) => {
            Promise.resolve(await l(event))),
```
If any of the listeners throw or return a rejected `Promise` that exception will unwrap all the way up to `Promise.all()` which will cause `Promise.reject()`, correct?

Reply:
It is correct and I added the unit test for it, not sure I understand your suggestion, the Promise.resolve will just make sure if the result is not a Promise to wrap it inside a Promise, and if it sends an exception or reject it is propagated by the Promise.all

Reply:
Which reminds me I forgot to add the test :)

Reply:
I did simplify it to `.map(async (l) => l(event)),`

Reply:
I had to disable the last part of my test as it runs locally fine but not in Github Actions. Its goal is to verify that the promise is not rejected with ensureDelivery=false, but it is normally sent to [EventTarget error handling](https://nodejs.org/api/events.html#events_eventtarget_error_handling)

Reply:
> It is correct and I added the unit test for it, not sure I understand your suggestion

Me neither anymore! LGTM :)